optuna/visualization/_regret_bound.py

```diff
@@ -0,0 +1,176 @@
+    trial_numbers: list[int]
+    regret_bounds: list[float]
+    errors: list[float] | None
+
+
+@experimental_func("3.2.0")
+def plot_regret_bound(
+    study: Study,
+    plot_error: bool = False,
+    improvement_evaluator: BaseImprovementEvaluator | None = None,
+    error_evaluator: BaseErrorEvaluator | None = None,
+    min_n_trials: int = DEFAULT_MIN_N_TRIALS,
+) -> "go.Figure":
+    _imports.check()
+
+    info = _get_regret_bound_info(study, plot_error, improvement_evaluator, error_evaluator)
+    return _get_regret_bound_plot(info, min_n_trials)
+
+
+def _get_regret_bound_info(
+    study: Study,
+    get_error: bool = False,
+    improvement_evaluator: BaseImprovementEvaluator | None = None,
+    error_evaluator: BaseErrorEvaluator | None = None,
+) -> _RegretBoundInfo:
+    if improvement_evaluator is None:
+        improvement_evaluator = RegretBoundEvaluator()
+    if error_evaluator is None:
+        error_evaluator = CrossValidationErrorEvaluator()
+
+    trial_numbers = []
+    regret_bounds = []
+    errors = []
+
+    for i, trial in enumerate(tqdm.tqdm(study.trials)):
+        trial_numbers.append(trial.number)
+        trials = study.trials[: i + 1]
+        trials = [t for t in trials if t.state == optuna.trial.TrialState.COMPLETE]
+
+        regret_bound = improvement_evaluator.evaluate(
+            trials=trials, study_direction=study.direction
+        )
+        regret_bounds.append(regret_bound)
+
+        if get_error:
+            error = error_evaluator.evaluate(trials=trials, study_direction=study.direction)
+            errors.append(error)
+
+    if len(errors) == 0:
+        return _RegretBoundInfo(
+            trial_numbers=trial_numbers, regret_bounds=regret_bounds, errors=None
+        )
+    else:
+        return _RegretBoundInfo(
+            trial_numbers=trial_numbers, regret_bounds=regret_bounds, errors=errors
+        )
+
+
+def _get_regret_bound_scatter(
+    trial_numbers: list[int],
+    regret_bounds: list[float],
+    opacity: float = 1.0,
+    showlegend: bool = True,
+) -> "go.Scatter":
+    plotly_blue_with_opacity = f"rgba(99, 110, 250, {opacity})"
+    return go.Scatter(
+        x=trial_numbers,
+        y=regret_bounds,
+        mode="markers+lines",
+        marker=dict(color=plotly_blue_with_opacity),
+        line=dict(color=plotly_blue_with_opacity),
+        name="Regret Bound",
+        showlegend=showlegend,
+        legendgroup="regret_bound",
+    )
+
+
+def _get_error_scatter(
+    trial_numbers: list[int],
+    errors: list[float] | None,
+) -> "go.Scatter":
+    if errors is None:
+        return go.Scatter()
+
+    plotly_red = "rgb(239, 85, 59)"
+    return go.Scatter(
+        x=trial_numbers,
+        y=errors,
+        mode="markers+lines",
+        name="Error",
+        marker=dict(color=plotly_red),
+        line=dict(color=plotly_red),
+    )
+
+
+def _get_reasonable_y_range(info: _RegretBoundInfo, min_n_trials: int) -> tuple[float, float]:
```

Comment:
I prefer `_get_y_range` for the function name, because "reasonable" is redundant and ambiguous. We'll expect a "reasonable" range even if the function name is `_get_y_range`.

What do you think?

Reply:
I see. I think `_get_y_range` is better by your reason, and also it is shorter. I'll fix it.